src/string_immutable.zig

```diff
@@ -5438,381 +5438,14 @@ pub fn pathContainsNodeModulesFolder(path: []const u8) bool {
-        0x1F947...0x1F9FF, //   [W] So   [185] FIRST PLACE MEDAL..NAZAR AMULET
-        0x1FA70...0x1FA7C, //   [W] So    [13] BALLET SHOES..CRUTCH
-        0x1FA80...0x1FA88, //   [W] So     [9] YO-YO..FLUTE
-        0x1FA90...0x1FABD, //   [W] So    [46] RINGED PLANET..WING
-        0x1FABF...0x1FAC5, //   [W] So     [7] GOOSE..PERSON WITH CROWN
-        0x1FACE...0x1FADB, //   [W] So    [14] MOOSE..PEA POD
-        0x1FAE0...0x1FAE8, //   [W] So     [9] MELTING FACE..SHAKING FACE
-        0x1FAF0...0x1FAF8, //   [W] So     [9] HAND WITH INDEX FINGER AND THUMB CROSSED..RIGHTWARDS PUSHING HAND
-        0x20000...0x2A6DF, //   [W] Lo [42720] CJK UNIFIED IDEOGRAPH-20000..CJK UNIFIED IDEOGRAPH-2A6DF
-        0x2A6E0...0x2A6FF, //   [W] Cn    [32] <reserved-2A6E0>..<reserved-2A6FF>
-        0x2A700...0x2B739, //   [W] Lo  [4154] CJK UNIFIED IDEOGRAPH-2A700..CJK UNIFIED IDEOGRAPH-2B739
-        0x2B73A...0x2B73F, //   [W] Cn     [6] <reserved-2B73A>..<reserved-2B73F>
-        0x2B740...0x2B81D, //   [W] Lo   [222] CJK UNIFIED IDEOGRAPH-2B740..CJK UNIFIED IDEOGRAPH-2B81D
-        0x2B81E...0x2B81F, //   [W] Cn     [2] <reserved-2B81E>..<reserved-2B81F>
-        0x2B820...0x2CEA1, //   [W] Lo  [5762] CJK UNIFIED IDEOGRAPH-2B820..CJK UNIFIED IDEOGRAPH-2CEA1
-        0x2CEA2...0x2CEAF, //   [W] Cn    [14] <reserved-2CEA2>..<reserved-2CEAF>
-        0x2CEB0...0x2EBE0, //   [W] Lo  [7473] CJK UNIFIED IDEOGRAPH-2CEB0..CJK UNIFIED IDEOGRAPH-2EBE0
-        0x2EBE1...0x2EBEF, //   [W] Cn    [15] <reserved-2EBE1>..<reserved-2EBEF>
-        0x2EBF0...0x2EE5D, //   [W] Lo   [622] CJK UNIFIED IDEOGRAPH-2EBF0..CJK UNIFIED IDEOGRAPH-2EE5D
-        0x2EE5E...0x2F7FF, //   [W] Cn  [2466] <reserved-2EE5E>..<reserved-2F7FF>
-        0x2F800...0x2FA1D, //   [W] Lo   [542] CJK COMPATIBILITY IDEOGRAPH-2F800..CJK COMPATIBILITY IDEOGRAPH-2FA1D
-        0x2FA1E...0x2FA1F, //   [W] Cn     [2] <reserved-2FA1E>..<reserved-2FA1F>
-        0x2FA20...0x2FFFD, //   [W] Cn  [1502] <reserved-2FA20>..<reserved-2FFFD>
-        0x30000...0x3134A, //   [W] Lo  [4939] CJK UNIFIED IDEOGRAPH-30000..CJK UNIFIED IDEOGRAPH-3134A
-        0x3134B...0x3134F, //   [W] Cn     [5] <reserved-3134B>..<reserved-3134F>
-        0x31350...0x323AF, //   [W] Lo  [4192] CJK UNIFIED IDEOGRAPH-31350..CJK UNIFIED IDEOGRAPH-323AF
-        0x323B0...0x3FFFD, //   [W] Cn [56398] <reserved-323B0>..<reserved-3FFFD>
-        => true,
-        else => false,
-    };
-}
-
 pub fn visibleCodepointWidth(cp: anytype) u3 {
     return visibleCodepointWidthType(@TypeOf(cp), cp);
 }
 
-pub fn visibleCodepointWidthType(comptime T: type, cp: T) usize {
-    if (isZeroWidthCodepointType(T, cp)) {
-        return 0;
-    }
-
-    if (isFullWidthCodepointType(T, cp)) {
-        return 2;
-    }
+extern "C" fn ghostty_simd_codepoint_width(cp: u32) u8;
 
-    return 1;
+pub fn visibleCodepointWidthType(comptime T: type, cp: T) usize {
+    return ghostty_simd_codepoint_width(@intCast(cp));
```

Comment:
this should be removed and `visibleCodepointWidth` should take `u32` instead of `anytype`. `u16`, `u21`, `u32` all coerce to `u32` with no cast necessary.